src/View/mainpage/StructureTab.py

```diff
@@ -409,18 +409,47 @@ def update_dict_polygons(self, state, roi_id):
             new_dict_polygons_coronal.pop(roi_name, None)
             new_dict_polygons_sagittal.pop(roi_name, None)
 
+    def on_rtss_selected(self, selected_rtss):
+        """
+        Function to run after a rtss is selected from SelectRTSSPopUp
+        """
+        self.patient_dict_container.get("existing_rtss_files").clear()
+        self.patient_dict_container.get("existing_rtss_files").append(
+            selected_rtss)
+        self.save_new_rtss()
+
+    def display_select_rtss_window(self):
+        """
+        Display a pop up window that contains all RTSSs attached to the
+        selected image set.
+        """
+        self.select_rtss_window = SelectRTSSPopUp(
+            self.patient_dict_container.get("existing_rtss_files"))
+        self.select_rtss_window.signal_rtss_selected.connect(
+            self.on_rtss_selected)
+        self.select_rtss_window.show()
+
     def save_new_rtss(self, event=None, auto=False):
         """
         Save the current RTSS stored in patient dictionary to the file system.
         :param event: Not used but will be passed as an argument from
         modified_indicator_widget on mouseReleaseEvent
         :param auto: Used for auto save without user confirmation
         """
-        if self.patient_dict_container.get("existing_file_rtss") is not None:
-            existing_rtss_directory = str(Path(self.patient_dict_container.get(
-                "existing_file_rtss")))
+        existing_rtss_files = self.patient_dict_container.get(
+            "existing_rtss_files")
+        if len(existing_rtss_files) == 1:
+            if isinstance(existing_rtss_files[0], Series):
+                self.existing_rtss_directory = str(Path(
+                    existing_rtss_files[0].get_files()[0]))
+            else:
+                self.existing_rtss_directory = existing_rtss_files[0]
+        elif len(existing_rtss_files) > 1:
+            self.display_select_rtss_window()
+            return
         else:
-            existing_rtss_directory = None
+            self.existing_rtss_directory = None
```

Comment:
why do we need self.existing_rtss_directory instead of existing_rtss_directory?

Reply:
Yea, you're right. It should be existing_rtss_directory. Fixed.